### PR TITLE
feat(payments): PAYPAL-903 Product Page Banners (Cornerstone) - creat…

### PR DESCRIPTION
**Pull request by bc-fetisov**
…e custom price event

#### What?

In order to show valid price in PDP with configured PayPal Commerce banner on it we need to create custom event that should be dispatched to initiate valid price or when product options changed.

#### Tickets / Documentation

- [https://developer.paypal.com/docs/integration/web/add-financing-banner/
- https://jira.bigcommerce.com/browse/PAYPAL-607

#### Screenshots (if appropriate)

https://www.loom.com/share/0d9b1c74e37d49c6862f547c8141cdaa

### CODE REVIEW

**Comment by bc-fetisov** [comment]
@bc-alexsaiannyi @BC-tymurbiedukhin could you review please?
cc @BC-ACherednichenko

File: assets/js/theme/common/utils/banner-utils.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { isBrowserIE } from './ie-helpers';
+
+const bannerUtils = {
+    dispatchProductBannerEvent: (productAttributes) => {
+        let price = 0;
+
+        if (!productAttributes.price || isBrowserIE) return;
```
Comment (bc-alexsaiannyi) [inline assets/js/theme/common/utils/banner-utils.js:7]: Don't you think it makes sense to switch these lines order - move if statement on line5?  if it's true we don't need to do extra action & create variable.

File: assets/js/theme/common/utils/banner-utils.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { isBrowserIE } from './ie-helpers';
+
+const bannerUtils = {
```
Comment (bc-alexsaiannyi) [inline assets/js/theme/common/utils/banner-utils.js:3]: what's the reason behind creating and importing object? Do expect to add more methods further?

**Comment by bc-alexsaiannyi** [comment]
@bc-fetisov, please update also changelog with your PR.

**Review by BC-ACherednichenko — APPROVED** [review]
LGTM

File: assets/js/theme/common/product-details.js [hunk 3]
```diff
@@ -181,6 +183,7 @@ export default class ProductDetails extends ProductDetailsBase {
             const productAttributesContent = response.content || {};
             this.updateProductAttributes(productAttributesData);
             this.updateView(productAttributesData, productAttributesContent);
+            bannerUtils.dispatchProductBannerEvent(response.data);
```
Comment (BC-tymurbiedukhin) [inline assets/js/theme/common/product-details.js:186]: could we use productAttributesData var here?

File: assets/js/theme/common/utils/banner-utils.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+        if (!productAttributes.price || isBrowserIE) return;
+
+        if (productAttributes.price.without_tax && !productAttributes.price.price_range) {
+            price = productAttributes.price.without_tax.value;
+        }
+
+        if (productAttributes.price.with_tax && !productAttributes.price.price_range) {
+            price = productAttributes.price.with_tax.value;
+        }
```
Comment (BC-tymurbiedukhin) [inline assets/js/theme/common/utils/banner-utils.js:15]: could we wrap this piece into `!productAttributes.price.price_range if statement`

File: assets/js/theme/common/utils/banner-utils.js (hunk 2 above)
Comment (bc-fetisov) [reply assets/js/theme/common/utils/banner-utils.js:3]: Yes, i do. I believe its quite possible that in the future we will need to handle different types of events related to banners.

File: assets/js/theme/common/utils/banner-utils.js (hunk 4 above)
Comment (bc-fetisov) [reply assets/js/theme/common/utils/banner-utils.js:15]: Ok, IMHO to much nested `lfs` reduce readability too, but for this case its doesn't matter

**Review by bc-alexsaiannyi — APPROVED** [review]

**Review by BC-tymurbiedukhin — APPROVED** [review]
LGTM

**Pull request merged**